ui/tests/acceptance/oidc-config/assignments-test.js

```diff
@@ -0,0 +1,197 @@
+import { module, test } from 'qunit';
+import { visit, currentURL, click, fillIn, findAll, currentRouteName } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import ENV from 'vault/config/environment';
+import authPage from 'vault/tests/pages/auth';
+import { OIDC_BASE_URL, SELECTORS } from 'vault/tests/helpers/oidc-config';
+import logout from 'vault/tests/pages/logout';
+import { create } from 'ember-cli-page-object';
+import fm from 'vault/tests/pages/components/flash-message';
+import {
+  clearRecord,
+  overrideCapabilities,
+  overrideMirageResponse,
+  ASSIGNMENT_LIST_RESPONSE,
+  ASSIGNMENT_DATA_RESPONSE,
+} from 'vault/tests/helpers/oidc-config';
+const flashMessage = create(fm);
+const ASSIGNMENTS_URL = OIDC_BASE_URL.concat('/assignments');
+
+// OIDC_BASE_URL = '/vault/access/oidc'
+
+module('Acceptance | oidc-config/assignments', function (hooks) {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  hooks.before(function () {
+    ENV['ember-cli-mirage'].handler = 'oidcConfig';
+  });
+
+  hooks.beforeEach(async function () {
+    this.store = await this.owner.lookup('service:store');
+    return authPage.login();
+  });
+
+  hooks.afterEach(function () {
+    return logout.visit();
+  });
+
+  hooks.after(function () {
+    ENV['ember-cli-mirage'].handler = null;
+  });
+
+  test('it renders only allow_all when no assignments are configured', async function (assert) {
+    assert.expect(3);
+
+    //* clear out test state
+    await clearRecord(this.store, 'oidc/assignment', 'test-assignment');
+
+    await visit(ASSIGNMENTS_URL);
+    assert.equal(currentURL(), '/vault/access/oidc/assignments');
+    assert
+      .dom('[data-test-oidc-assignment-linked-block]')
+      .includesText('allow_all', 'displays default assignment');
+    assert.dom('[data-test-oidc-assignment-linked-block]').hasClass('is-disabled', 'Allow all is disabled.');
+  });
+
+  test('it creates, updates, and deletes an assignment', async function (assert) {
+    assert.expect(12);
+    await visit(ASSIGNMENTS_URL);
+    // create a new assignment
+    await click(SELECTORS.assignmentCreateButton);
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.oidc.assignments.create',
+      'navigates to create form in empty state'
```

Comment:
since there's no empty state for assignments, we're just navigating from the create button, right?

Reply:
ah yep, let me amend the statement.